local-tests/tests/testSignTransactionWithSolanaEncryptedKey.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  LitAccessControlConditionResource,
+  LitActionResource,
+  createSiweMessageWithRecaps,
+  generateAuthSig,
+} from '@lit-protocol/auth-helpers';
+
+/**
+ * Test Commands:
+ * ✅ NETWORK=cayenne yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ * ✅ NETWORK=manzano yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ * ❔ NETWORK=localchain yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ */
+export const testSignTransactionWithSolanaEncryptedKey = async (
+  devEnv: TinnyEnvironment
+) => {
+  const alice = await devEnv.createRandomPerson();
+  const solanaKeypair = Keypair.generate();
+  const decryptionAccessControlConditions: AccessControlConditions = [
+    {
+      contractAddress: '',
+      standardContractType: '',
+      chain: 'ethereum',
+      method: '',
+      parameters: [':userAddress', 'latest'],
+      returnValueTest: {
+        comparator: '=',
+        value: await alice.wallet.getAddress(),
+      },
+    },
+  ];
+
+  const { ciphertext, dataToEncryptHash } = await encryptString(
+    {
+      accessControlConditions: decryptionAccessControlConditions,
+      dataToEncrypt: Buffer.from(solanaKeypair.secretKey).toString('hex'),
+    },
+    devEnv.litNodeClient as unknown as ILitNodeClient
+  );
+
+  const sessionSigs = await devEnv.litNodeClient.getSessionSigs({
+    chain: 'ethereum',
+    resourceAbilityRequests: [
+      {
+        resource: new LitActionResource('*'),
+        ability: LitAbility.LitActionExecution,
+      },
+      {
+        resource: new LitAccessControlConditionResource('*'),
+        ability: LitAbility.AccessControlConditionDecryption,
+      },
+    ],
+    authNeededCallback: async ({ uri, resourceAbilityRequests }) => {
+      const toSign = await createSiweMessageWithRecaps({
+        uri: uri!,
+        expiration: new Date(Date.now() + 1000 * 60 * 10).toISOString(),
+        resources: resourceAbilityRequests!,
+        walletAddress: await alice.wallet.getAddress(),
+        nonce: await devEnv.litNodeClient!.getLatestBlockhash(),
+        litNodeClient: devEnv.litNodeClient,
+      });
+      return await generateAuthSig({
+        signer: alice.wallet,
+        toSign,
+      });
+    },
+  });
+
+  const solanaTransaction = new Transaction();
+  solanaTransaction.add(
+    SystemProgram.transfer({
+      fromPubkey: solanaKeypair.publicKey,
+      toPubkey: new PublicKey(solanaKeypair.publicKey),
+      lamports: LAMPORTS_PER_SOL / 100, // Transfer 0.01 SOL
+    })
+  );
+  solanaTransaction.feePayer = solanaKeypair.publicKey;
+
+  const solanaConnection = new Connection(clusterApiUrl('devnet'), 'confirmed');
+  const { blockhash } = await solanaConnection.getLatestBlockhash();
+  solanaTransaction.recentBlockhash = blockhash;
+
+  const serializedTransaction = solanaTransaction
+    .serialize({
+      requireAllSignatures: false,
+      verifySignatures: false,
+    })
+    .toString('base64');
```

Comment:
Can we do this inside the `signTransactionWithSolanaEncryptedKey` litAction as the user is expected to provide only an unsigned transaction in the interface `SolanaLitTransaction`? The idea is to model is in a similar approach as `signTransactionWithEncryptedKey`

Reply:
No I don't think we should. As mentioned [here](https://github.com/LIT-Protocol/js-sdk/pull/504/files#r1640566254) Solana txs aren't just simple objects like ETH txs regardless of what you're trying to do on chain. I believe it's easiest for the user to use Solana's web3.js library to create their specific transaction instance (with all the Solana instructions), and then serialize it and provide it to the Lit Action for signing

Reply:
Based on my response: https://github.com/LIT-Protocol/js-sdk/pull/504#discussion_r1641997975
How about us setting the other params as default inside the Lit Action?

Reply:
Given how many different types of transactions there are in the Solana SDK, I still think we should just have the user create and serialize their tx, then pass it to the LA for signing

Reply:
`requireAllSignatures ` & `verifySignatures ` should be false to submit an unsigned tx. Otherwise, it needs to be signed by the user

Reply:
Using serialized Tx for Solana